AdminPortal/src/pages/dashboard.tsx

```diff
@@ -1,6 +1,285 @@
+}
+
+export default function App() {
+  const [activeTab, setActiveTab] = useState<string>('Needs Review');
+
+  const [selectAllClicked, setSelectAllClicked] = useState(
+    new Array(4).fill(false)
+  );
+
+  // for the dropdown
+  const [selected, setSelected] = useState<Option | null>(null);
+
+  // mapping the tabs to an integer key
+  const tabMapping = new Map();
+  tabMapping.set('Needs Review', 1);
+  tabMapping.set('In Review', 2);
+  tabMapping.set('Verified', 3);
+  tabMapping.set('Disapproved', 4);
+
+  const [items, setItems] = useState<DataItem[]>([
+    {
+      id: 1,
+      dateApplied: '2022-01-01',
+      name: 'Hello 1',
+      mobileNumber: '+1 123 456 7890',
+      licenseID: 'A1234567',
+      licensePlate: 'ABC-123',
+      isChecked: false,
+      category: 'Needs Review',
+    },
+    {
+      id: 2,
+      dateApplied: '2022-02-01',
+      name: 'Hello 2',
+      mobileNumber: '+1 987 654 3210',
+      licenseID: 'B2345678',
+      licensePlate: 'DEF-456',
+      isChecked: false,
+      category: 'In Review',
+    },
+    {
+      id: 3,
+      dateApplied: '2022-03-01',
+      name: 'Hello 3',
+      mobileNumber: '+1 111 222 3333',
+      licenseID: 'C3456789',
+      licensePlate: 'GHI-789',
+      isChecked: false,
+      category: 'Verified',
+    },
+    {
+      id: 4,
+      dateApplied: '2022-01-01',
+      name: 'Hello 4',
+      mobileNumber: '+1 123 456 7890',
+      licenseID: 'A1234567',
+      licensePlate: 'ABC-123',
+      isChecked: false,
+      category: 'Needs Review',
+    },
+    {
+      id: 5,
+      dateApplied: '2022-02-01',
+      name: 'Hello 5',
+      mobileNumber: '+1 987 654 3210',
+      licenseID: 'B2345678',
+      licensePlate: 'DEF-456',
+      isChecked: false,
+      category: 'Disapproved',
+    },
+  ]);
+
+  const options = [
+    { value: 'Needs Review', label: 'Needs Review' },
+    { value: 'In Review', label: 'In Review' },
+    { value: 'Verified', label: 'Verified' },
+    { value: 'Disapproved', label: 'Disapproved' },
+  ];
+
+  const customStyle = {
+    control: (styles: ControlStyles) => ({
+      ...styles,
+      borderColor: '#9e9e9e',
+      height: '100%',
+      width: '260px',
+    }),
+  };
+
+  const handleTabClick = (tab: string) => {
+    setActiveTab(tab);
+  };
+
+  const handleDropdownClick = (selectedOption: Option) => {
+    setSelected(selectedOption);
+
+    setItems(
+      items.map((item) =>
+        item.isChecked === true
+          ? { ...item, isChecked: false, category: selectedOption.label }
+          : item
+      )
+    );
+
+    setSelectAllClicked(new Array(4).fill(false));
+  };
+
+  const handleSelectAll = (): void => {
+    setSelectAllClicked((prevState: Array<boolean>) => {
+      let indexToFlip = tabMapping.get(activeTab);
+
+      // convering to 0 based indexing
+      indexToFlip = indexToFlip - 1;
+
+      console.log(indexToFlip);
+
+      let newState: boolean[] = [];
+
+      for (let i = 0; i < 4; i++) {
```

Comment:
again, let's try to set this to a constant or use the inherent length of whatever we're iterating through

Reply:
fixed